src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1510,51 +1532,42 @@ private String quoteUnixCredentials(String str) {
         return str.replace("'", "'\\''");
     }
 
-    private File createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOException {
-        File ssh = File.createTempFile("pass", ".bat");
-        try (PrintWriter w = new PrintWriter(ssh)) {
-            w.println("echo \"" + quoteWindowsCredentials(Secret.toString(sshUser.getPassphrase())) + "\"");
-            w.flush();
-        }
-        ssh.setExecutable(true);
+    private FilePath createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOException, InterruptedException {
+        FilePath ssh = workspace.createTempFile("pass", ".bat");
+        ssh.write("echo \"" + Secret.toString(sshUser.getPassphrase()) + "\"", null);
+        ssh.chmod(0700);
```

Comment:
I'm not sure that we want the unencrypted ssh passphrase written into the workspace.  Isn't that even more widely available than the system temporary directory (through the "Workspace" browsing facilities in Jenkins).

I'm considering switching from `echo` to `type` and `cat` so that the shell escaping rules of echo don't have to be applied to the passphrase.  Refer to https://github.com/MarkEWaite/git-client-plugin/pull/2/commits/22ae93a9f5d1088fb353995e69bda2cb0927c839 for the beginnings of that code.

I'd rather have this change (modify the location of the passphrase) be considered in the context of #207, since it is also touching the same area, but touching it in a way that fixes a Windows problem.